server/server.go

```diff
@@ -74,11 +74,13 @@ func attemptTcpConnection(port int, opts *options.Options) error {
 
 	defaultTimeout := time.Second * 5
 
-	_, err := net.DialTimeout("tcp", fmt.Sprintf("0.0.0.0:%d", port), defaultTimeout)
+	conn, err := net.DialTimeout("tcp", fmt.Sprintf("0.0.0.0:%d", port), defaultTimeout)
 	if err != nil {
 		return err
 	}
 
+	defer conn.Close()
```

Comment:
There should be an automated test that checks both success and failure conditions for the health checker. That might've caught this issue.

Reply:
Agreed this would def benefit from automated tests (!) but it wouldn't have caught this particular issue. What was happening was that the number of ongoing checks kept leaving files open on the local file system (each of which represented an open network connection). Eventually, we hit I believe the "file descriptors limit". The tests only would have caught this if we had simulated hundreds or thousands of health-check attempts.